### ISSUE TITLE
Python code not working

### ISSUE THREAD

**Issue by tony1tf** [MaxusDev/MWTool#1]
The unit sends its configuration data  at bootup according to Coolterm, but the Python code doesn't send anything to trigger raw mode, so just hangs when trigger is pressed.
The serial read needs a timeout because <ctrl>C struggles to terminate the read. 
Python error when pressing the unfinished option IR.

**Comment by tony1tf** [comment]
This is what the module sends on bootup:
->> Maxustech Inc,. 
|->> Reaserch Support Center - Nov 2019 Ver1.1 
|->> Drive MCU 76E003 
|->> PA: 1 REVGAIN: 7 THRES: 30 FREQ: 0 DEBUG: 0 
|->> LIGHT: 0 DELAY: 2
Can you confirm this is the latest firmware, please. 
What commands are there to upload new firmware, and to reset the module?

**Comment by MaxusDev** [comment]
> This is what the module sends on bootup:
> ->> Maxustech Inc,.
> |->> Reaserch Support Center - Nov 2019 Ver1.1
> |->> Drive MCU 76E003
> |->> PA: 1 REVGAIN: 7 THRES: 30 FREQ: 0 DEBUG: 0
> |->> LIGHT: 0 DELAY: 2
> Can you confirm this is the latest firmware, please.
> What commands are there to upload new firmware, and to reset the module?

The communication protocol has been updated. The device will set to normal mode by default which doesn't enable raw data output. Use AT+DEBUG=0002  to set it to debug mode.

**Comment by tony1tf** [comment]
You have closed this  issue, but not fixed the problem. By saying 'The communication protocol has been updated', are you saying that the latest devices have firmware which doesn't enable raw data by default? If so, then you must update your Python code to enable raw data when the trigger button is pressed. In any case, I have experimented with 'AT+DEBUG=0002' from the terminal, and never seen any raw data from the device. You have also not answered my question about how new firmware can be uploaded.
